### PR TITLE
Add support for CockroachDB StateStore

### PR DESCRIPTION
**Pull request by r-c-correa**
## Description
Add Query support for  CockroachDB StateStore

## Issue reference
#2236

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+description: Detailed information on the CockroachDB state store component
+aliases:
+  - "/operations/components/setup-state-store/supported-state-stores/setup-cockroachdb/"
+---
+
+## Create a Dapr component
+
+Create a file called `cockroachdb.yaml`, paste the following and replace the `<CONNECTION STRING>` value with your connection string. The connection string for CockroachDB follow the same standard for PostgreSQL connection string. For example, `"host=localhost user=root port=26257 connect_timeout=10 database=dapr_test"`. See the CockroachDB [documentation on database connections](https://www.cockroachlabs.com/docs/stable/connect-to-the-database.html) for information on how to define a connection string.
+
+If you want to also configure CockroachDB to store actors, add the `actorStateStore` configuration element shown below.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md]: Given that the overview describes that this state store supports actors, you do not need to repeat this here. Suggest removing the actor examples.

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+```yaml
+  - name: actorStateStore
+    value: "true"
+```
+
+
+## Create CockroachDB
+
+{{< tabs "Self-Hosted" >}}
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md]: Can you also provide an example of how to deploy to Kubernetes? Also any other hosting environments if these are easy to reference into the CockroachDB docs (e.g. Deploying to production VM environment)

**Comment by msfussell** [comment]
@r-c-correa PTAL at my suggested changes.

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md (hunk 1 above)
Comment (r-c-correa) [reply daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md]: Of course, I'll make the adjustment 😄

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md (hunk 2 above)
Comment (r-c-correa) [reply daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md]: Of course, I'll make the adjustment 😃

**Review by r-c-correa — COMMENTED** [review]
I made changes in another commit. 😢

**Comment by msfussell** [comment]
@r-c-correa - You have pulled in several other changes to the PR. Can you close this PR and open a new ones (or changes this one) to have just the DB Index file and the CockroachDB file. I am unclear on your changes here otherwise.

**Comment by r-c-correa** [comment]
I create a new [Pull Request](https://github.com/dapr/docs/pull/2276) @msfussell 😄

**Pull request closed without merging**